Migrate test-integration-cli experimental plugin tests to integration

vdemeester opened the pull request:
All `plugins` tests that require an `ExperimentalDaemon` are migrated
to `integration/plugin/*` and start an experimental daemon to test on
it.

The end goal being to remove the `experimental` build.

:lion: 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

cpuguy83 commented on integration/plugin/graphdriver/external_test.go:
```diff
@@ -330,77 +345,83 @@ func (s *DockerExternalGraphdriverSuite) setUpPlugin(c *check.C, name string, ex
+
+			ctx := context.Background()
+
+			/*
```
??

vdemeester replied:
oups, forgot to remove that.. :sweat_smile:

cpuguy83 commented on integration/plugin/graphdriver/overlay2_test.go:
```diff
@@ -0,0 +1,51 @@
+	ctx := context.Background()
+
+	// install the plugin
+	plugin := "cpuguy83/docker-overlay2-graphdriver-plugin"
```
We should build this locally instead of pulling.

vdemeester replied:
Probably :stuck_out_tongue:

anusha-ragunathan commented on integration/plugin/graphdriver/overlay2_test.go:
```diff
@@ -0,0 +1,51 @@
+	})
+	defer responseReader.Close()
+	assert.NilError(t, err)
+	// ensure it's done by waiting for EOF on the response
```
Curious to know why we care to read the response here, while in most others we assume a successful pull/install and io.Discard the response.

vdemeester replied:
oh yeah, it should be `io.Copy` with `ioutil.Discard` as destination

anusha-ragunathan commented:
Plugin deployment in swarm mode is also experimental. For eg, `TestAPISwarmServicesPlugin`

vdemeester commented on integration/plugin/graphdriver/overlay2_test.go:
```diff
@@ -0,0 +1,51 @@
+
+	// restart the daemon with the plugin set as the storage driver
+	d.Stop(t)
+	d.StartWithBusybox(t, "-s", plugin, "--storage-opt", "overlay2.override_kernel_check=1")
```
i.e run a container, see the diff, etc ?

anusha-ragunathan replied:
yes

vdemeester replied:
it's now done 😉

vdemeester committed:
Migrate test-integration-cli experimental plugin tests to integration

All `plugins` tests that require an `ExperimentalDaemon` are migrated
to `integration/plugin/*` and start an experimental daemon to test on
it.

The end goal being to remove the `experimental` build.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

cpuguy83 reviewed (approved):
LGTM
